server/src/main/java/org/opensearch/index/shard/StoreRecovery.java

```diff
@@ -397,7 +400,8 @@ void recoverFromSnapshotAndRemoteStore(
 
                 RemoteSegmentStoreDirectoryFactory directoryFactory = new RemoteSegmentStoreDirectoryFactory(
                     () -> repositoriesService,
-                    threadPool
+                    threadPool,
+                    remoteStoreSettings
```

Comment:
this method is just called at a single place in IndexShard. we can pass the prefix/remote store settings just to this function instead of change the contract for the constructor which is leading to changes to all the places where StoreRecovery is initializing

Reply:
newDirectory is part of DirectoryFactory interface which comes from the IndexStorePlugin plugin. I dont see changing this method as an option for many reasons. The information needs to be brought in as part of creation only.

Reply:
let me clarify, i feel my previous comment should have been added at the starting of `recoverFromSnapshotAndRemoteStore` method. 

what i mean is since `recoverFromSnapshotAndRemoteStore` is remote store specific function, we can just pass the `remoteStoreSettings` object to `recoverFromSnapshotAndRemoteStore` instead of bloating the ctor of `StoreRecovery` with remote store specific args.